internal/ingress/controller/nginx.go

```diff
@@ -327,6 +350,22 @@ func (n *NGINXController) Start() {
 		}()
 	}
 
+	if n.gRPCServer != nil {
+		klog.InfoS("Starting grpc server", "port", n.cfg.ListenPorts.GRPCPort)
+		ingress.RegisterEventServiceServer(n.gRPCServer, &EventServer{Recorder: n.recorder})
+		ingress.RegisterConfigurationServer(n.gRPCServer, &ConfigurationServer{
+			n: n,
+		})
+
+		lis, err := net.Listen("tcp", fmt.Sprintf("0.0.0.0:%d", n.cfg.ListenPorts.GRPCPort))
+		if err != nil {
+			klog.Fatalf("failed to listen: %v", err)
+		}
+		go func() {
+			klog.ErrorS(n.gRPCServer.Serve(lis), "failed to start gRPC Server")
+		}()
```

Comment:
I think it is enough to print the log directly? Don't need goroutine

Reply:
Inside the goroutine there is a Serve directive. If I don't wait for it, it will wait until error and I cannot register the servers as below